### DISCUSSION TITLE
SM200 storage transfer system with buffer (J)

### DISCUSSION THREAD

**Discussion by volklend** [emsesp/EMS-ESP32#2212] [Ideas]
I have an  solar system with module SM200 and transfer system with buffer (chapter 2.3.10 in the german installation manual).
The pump is driven by PS7 (PS4 in Hardware) an the Temperatures are measured by TS10 and TS11 (TS7 and TS8 in Hardware).
It would be great to have these entity's in the module list.
![SM200 - !AEJ (BP)](https://github.com/user-attachments/assets/89251d0d-8bbb-48dc-9835-85b5ef67bb2e)

**Comment by MichaelDvP** [comment]
The TS numbering from Bosch is always confusing. 
For the temperatures i expect them in telegram 0x0362, actual we don't know the offsets 4, 12, 14, 18, 22, 24
In telnet or 3.7.1 web status->logger use `read 30 362` and compare the searched temperature with a 16bit value, `80 00` means no sensor connected, other values are factor 10, so `02 00` means 512 -> 31,2°C.
If you thinks you have identified the right offset, create a custom entity with device 0x30, type 0x362, offset ?, INT16 and monitor the changing values.
For the pump i expect telegram 0x036A, unknown offsets are 3, 5, 6, 8, 9, 11. Value 03 means off, 04 means on. Use `read 30 36A` when  pump is running and second time when pump is stopped, compare telegrams to find the right offset.

If you need help: Set status-logger to log all, leave page open, got to heating and note the missing temperatures, stop logger after 5 minutes and download log. Attach log here and tell us the temperatures you have noted while logging.
For the pump make a1-2 minute  log while pump is running and another while pump is stopped.

**Reply by volklend** [reply]
I have done a first logging and returned the logfile an a picture with temperatures. Perhaps you can help me and indentify the  entities for  me, this is Day 2 with EMS-ESP for me...
![Umladung](https://github.com/user-attachments/assets/e8b79c00-ccd5-4f0b-af75-26e7ad12327f)
[log.txt](https://github.com/user-attachments/files/17790454/log.txt)
[emsesp_system_info.json](https://github.com/user-attachments/files/17790455/emsesp_system_info.json)

**Reply by volklend** [reply]
The second logging is pump on/off
![pump_on](https://github.com/user-attachments/assets/2ac080e7-861c-4393-b97d-163553265993)
[pump_on.txt](https://github.com/user-attachments/files/17790561/pump_on.txt)
![pump_off](https://github.com/user-attachments/assets/354eeb4b-ee96-456c-b125-02487579a22e)
[pump_off.txt](https://github.com/user-attachments/files/17790562/pump_off.txt)

**Reply by volklend** [reply]
![image](https://github.com/user-attachments/assets/1e6da9dc-d5c8-494f-aeab-083f9f6282ca)
I could identify with your information TS10 and TS11. 😊
So it is eventually possible to read TS6. Then I can connect one more sensor to monitor my heat-exchanger.

**Reply by MichaelDvP** [reply]
Great, i'll add TS10/TS11 to the solar. 
The transfer pump seems to be 0x36A, offset 11, for on/off state (04-on, 03-off), INT8 
and pump modulation in 0x364, offset 14, INT8.
Can you test and confirm.
> So it is eventually possible to read TS6.

Try it, i don't know how the module handles the other inputs.

**Reply by volklend** [reply]
Hi Michael,
I have tested pump on/off and modulation. It's working...
![PS7_on](https://github.com/user-attachments/assets/42bbee66-57e7-4ce4-a1dc-3d198980470d)
![PS7_off](https://github.com/user-attachments/assets/3249f912-109a-44e8-b16c-298fc434498a)

Now I'm on my way to replace my old "Citrin Solar SLR XT" system with pictures:
https://www.vbus.net/LJR-Ld
and configrable diagrams
https://www.vbus.net/DJR-2T0

Maybe NodeRed is a suitable workbench ?

**Reply by volklend** [reply]
Hi Michael,
Thank you very much für implementing.

In the between time I have organized a RaspBerryPi  and installed
mosquitto, MQTT Explore, Telegraf, InfluxDB, Grafana, ioBroker with adapters for Homematic, KNX, EMS-ESP  and Java. Quite hard work, but very interesting. And replaces perfect my oldscool Citrin Solar system. 

I found one entity but no Data in the protokols. Do you have Information about "Solareinfluss" ?

![20241229_221155](https://github.com/user-attachments/assets/5036d8a0-a347-484f-8731-c89154874012)
![20241229_221145](https://github.com/user-attachments/assets/e3af0570-a784-46c6-b843-c7b18d2b40ed)

**Reply by MichaelDvP** [reply]
> Do you have Information about "Solareinfluss" ?

No, just log the telegrams and change the setting to different values, we should see this in the log.

**Reply by volklend** [reply]
The Command Value is in 0x2AF Offset 3 INT8  from -1 to -5. 0 is "Disabled". 
![image](https://github.com/user-attachments/assets/539ad335-8d9a-4b21-a960-a722947eb688)
So in the moment only the diagnostic value adress is unknown. And without your List of unknown temperature adresses, it is quite hard to find...

![20241231_141702](https://github.com/user-attachments/assets/e8e0e9e0-ff53-425e-b4e3-8c5c1d361b58)

**Comment by Sbried** [comment]
Maybe you find your -1,1K in 0x2A5 offset 29. It´s int8 with 0,1 K resolution

**Reply by volklend** [reply]
Yes, i have tested and it works ! 👍 
![image](https://github.com/user-attachments/assets/7ddc2183-7533-4a4a-8ee1-111ed38b35ee)
![20250101_130021](https://github.com/user-attachments/assets/53461e88-0401-4468-beb6-6a2ad91b9f37)

Happy new year everybody. :
![NewYearWishes2024HappyNewYear2024WishesGIF](https://github.com/user-attachments/assets/40811967-c5ac-431b-9d85-f2ca14365f3b)

**Reply by Sbried** [reply]
Maybe @MichaelDvP can add data type 0x2A5 offset 29 as solar influence to the code?

**Reply by volklend** [reply]
Der Solareinfluss ist evtl. unterschiedlich zwischen hc1 und hc2. Die hc2-Adressen habe ich noch nicht, könnte ich aber sobald das wetter besser ist jetzt mit meinem zweiten EMS-ESP an meiner anderen Anlage testen...

**Reply by volklend** [reply]
Gibt es eine Prognose für die offsets ?

**Reply by Sbried** [reply]
HC2 data is at data type 0x02A6 (HC3 0x02A7 etc, ...). Offsets are identical as for 0x02A5

**Reply by volklend** [reply]
thank you very much for this information. I will test as soon as the sun shines again 😊

**Reply by volklend** [reply]
I am thinking about creating my own room influence and calculate a own value for tempautotemp.
Do you think, i can write this item very often without destroying any EEPROM ?

**Reply by volklend** [reply]
![image](https://github.com/user-attachments/assets/b5b0fbaf-9242-49d5-8a00-14a4c18e3cb6)
The adress/offset is correct. The  current values are scaled in 0.1C.
